File: data_structures/heap/max_heap/find_k_largest.cpp
// Note: C++ std::priority_queue implements a max-heap. To achieve min-heap functionality, we will multiply the values by -1 before pushing them onto the heap.
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int> heap;
        for (int num: nums) {
            heap.push(-num);
            if (heap.size() > k) {
                heap.pop();
            }
        }
        
        return -heap.top();
    }
};

File: heap/max_heap/find_k_largest.rs
use std::collections::BinaryHeap; 
// max heap is default in binary collections
impl Solution {
    pub fn find_kth_largest(nums: Vec<i32>, k: i32) -> i32 {
        let mut heap = BinaryHeap::from(nums);
        let mut counter = k;
        while counter > 1 {
            heap.pop();
            counter -= 1;
        }
        *heap.peek().unwrap()
    }
}